File: 0682-baseball-game/0682-baseball-game.ts
function calPoints(operations: string[]): number {
    let st = [];

    for (let i=0; i<operations.length; i++) {
        if (!isNaN(Number(operations[i]))) 
            st.push(Number(operations[i]))
        else if (operations[i] === '+') {
            let p1 = st.pop();
            let p2 = st.pop();
            let res = p1+p2;
            st.push(p2);
            st.push(p1);
            st.push(res);
        } else if (operations[i] === 'D')
            st.push(st[st.length-1] * 2);
        else if (operations[i] === 'C')
            st.pop();
    }

    return st.reduce((acc, curr) => curr+acc, 0);
};

File: 0682-baseball-game/0682-baseball-game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        st = []
        res = 0

        for i in range(len(operations)):
            if operations[i]=="C":
                st.pop()
            elif operations[i]=="D":
                a = st[-1]*2
                st.append(a)
            elif operations[i]=="+":
                if len(st)>=2:
                    a = st.pop()
                    b = st.pop()
                    st.append(b)
                    st.append(a)
                    st.append(a+b)
            else:
                st.append(int(operations[i]))
        
        for i in range(len(st)): res+=st[i]

        return res